### ISSUE TITLE
Documentaion for CNTKLib.ForwardBackward in C#

### ISSUE THREAD

**Issue by HenrikOjelund** [Microsoft/CNTK#3332]
I am working with CNTK from C#. We are testing using a neural net as part of a loss function. Hence, we need to calculate the gradient with the respect to the input. I can see from the CNTK: 205 example, that this is straight forward using Python. Can I use the ForwardBackward method for this?

**Comment by KeDengMS** [comment]
I think you are looking for Function::Forward and Function::Backward [here](https://github.com/Microsoft/CNTK/blob/master/Source/CNTKv2LibraryDll/API/CNTKLibrary.h#L3233-L3266). These two would provide gradient computation. The Python functions in Tutorial 205 are wrappers on top of them.

**Comment by HenrikOjelund** [comment]
Yes, that is what I need. How would I wrap these functions to C#? Do I have to build my own custom CNTKLib?

**Comment by KeDengMS** [comment]
Seems [those functions](https://github.com/Microsoft/CNTK/blob/master/bindings/common/CNTKManagedCommon.i#L197-L198) are not exposed in C# yet.

**Comment by HenrikOjelund** [comment]
Thanks, that would really help us. Do you know if it can make it for version 2.6?

**Comment by GuntaButya** [comment]
Any update on this fellas?

> Seems those functions are not exposed in C# yet.
